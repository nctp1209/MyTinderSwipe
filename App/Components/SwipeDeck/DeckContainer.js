import React, { Component } from "react";
import { View } from "react-native";
import PropTypes from "prop-types";

import SwipeCard from "./SwipeCard";

export default class DeckContainer extends Component {
  static defaultProps = {
    listCard: []
  };

  static propTypes = {
    listCard: PropTypes.array,
    renderCardItem: PropTypes.func
  };

  render() {
    if (this.props.listCard && this.props.listCard.length > 0) {
      const listComponent = [];
      this.props.listCard.forEach((item, i) => {
        listComponent.push(
          <SwipeCard
            key={i}
            index={i}
            onSwipeLeft={() => this.props.onSwipeLeft(item)}
            onSwipeRight={() => this.props.onSwipeRight(item)}
            renderCardItem={() => this.props.renderCardItem(item)}
          />
        );
      });
      return listComponent.reverse();
    }
    return <View />;
  }
}
