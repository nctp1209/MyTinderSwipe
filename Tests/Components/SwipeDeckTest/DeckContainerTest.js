import React from "react";
import { Text } from "react-native";
import renderer from "react-test-renderer";
import { path } from "ramda";
import { shallow } from 'enzyme'

import APIResult from "../../APIResult";
import DeckContainer from "../../../App/Components/SwipeDeck/DeckContainer";

test("DeckContainer render when list card is empty", () => {
  const tree = renderer
    .create(
      <DeckContainer
        listCard={[]}
        renderCardItem={item => <Text>{item.user.username}</Text>}
        onSwipeLeft={item => console.log("onSwipeLeft")}
        onSwipeRight={item => console.log("onSwipeRight")}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("DeckContainer render when list card have data", () => {
  const response = APIResult.getPeople();
  const listUser = path(["data", "results"], response);
  const tree = renderer
    .create(
      <DeckContainer
        listCard={listUser}
        renderCardItem={item => <Text>{item.user.username}</Text>}
        onSwipeLeft={item => console.log("onSwipeLeft")}
        onSwipeRight={item => console.log("onSwipeRight")}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
