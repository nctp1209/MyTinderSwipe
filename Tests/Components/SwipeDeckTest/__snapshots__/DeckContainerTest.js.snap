// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckContainer render when list card have data 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 5,
      "height": 1134,
      "margin": 20,
      "position": "absolute",
      "transform": Array [
        Object {
          "rotate": "0deg",
        },
        Object {
          "translateX": 0,
        },
        Object {
          "translateY": 0,
        },
      ],
      "width": 710,
    }
  }
>
  <Text>
    tinymeercat81
  </Text>
</View>
`;

exports[`DeckContainer render when list card is empty 1`] = `<View />`;
